src/compiler/emitter.ts

```diff
@@ -3054,7 +3054,31 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                     node.parent.kind === SyntaxKind.SourceFile;
             }
 
+            function filterNonInitializedExportedVariableDeclarations(nodes: NodeArray<VariableDeclaration>, flag: NodeFlags): NodeArray<VariableDeclaration> {
+                let result = <NodeArray<VariableDeclaration>>Object.create(nodes);
+                result.splice(0);
+                for (let i = 0; i<nodes.length; i++) {
+                    if (!(nodes[i].parent.parent.flags & NodeFlags.Export && !nodes[i].initializer)) {

```

Comment:
This requires climbing up the parent references twice for every variable; you can just move the Export check to the VariableStatement node like I originally suggested, and then call filter as I describe below.